dev/tools/gen_date_localizations.dart

```diff
@@ -125,7 +125,7 @@ String _jsonToMap(dynamic json) {
   if (json is Iterable)
     return 'const <dynamic>[${json.map(_jsonToMap).join(',')}]';
 
-  if (json is Map) {
+  if (json is Map<String, dynamic>) {
```

Comment:
Is a `Map<String, dynamic>` not a `Map`?

Reply:
By testing against `Map<String, dynamic>`, the type of `json` gets promoted so that in the `json.forEach` below, the type checker knows that it's ok to assume that `key` will be a string. 

Testing against `Map` doesn't guarantee that the keys will be strings, and so the closure below would have to be prepared to take any key.

Reply:
Ah, cunning!